djangoappengine/storage.py

```diff
@@ -50,63 +73,106 @@ def serve_file(request, file, save_as, content_type, **kwargs):
     return response
```

Comment:
I haven't tested this, I don't have this usecase in my application. There may be some subtlety that I missed. Or are you using the latest cloudstorage library?